internal/services/healthcare/healthcare_dicom_resource.go

```diff
@@ -373,3 +532,103 @@ func flattenDicomServicePrivateEndpoint(input *[]dicomservices.PrivateEndpointCo
 	}
 	return results
 }
+
+func expandStorageConfiguration(input []interface{}) *dicomservices.StorageConfiguration {
+	if len(input) == 0 || input[0] == nil {
+		return nil
+	}
+	storageSettings := input[0].(map[string]interface{})
+
+	var storageResourceId *string
+	if v, ok := storageSettings["storage_account_id"]; ok {
+		storageResourceId = pointer.To(v.(string))
+	}
+
+	var fileSystemName *string
+	if v, ok := storageSettings["file_system_name"]; ok {
+		fileSystemName = pointer.To(v.(string))
+	}
+
+	return &dicomservices.StorageConfiguration{
+		FileSystemName:    fileSystemName,
+		StorageResourceId: storageResourceId,
+	}
+}
+
+func flattenStorageConfiguration(configuration *dicomservices.StorageConfiguration) interface{} {
+	if configuration == nil {
+		return []interface{}{}
+	}
+
+	result := make(map[string]interface{})
+	if configuration.FileSystemName != nil {
+		result["file_system_name"] = *configuration.FileSystemName
+	}
+
+	if configuration.StorageResourceId != nil {
+		result["storage_account_id"] = *configuration.StorageResourceId
+	}
+
+	return []interface{}{result}
+}
+
+func expandDicomServiceCorsConfiguration(inputList []interface{}) *dicomservices.CorsConfiguration {
+	if len(inputList) == 0 {
+		return nil
+	}
+
+	input := inputList[0].(map[string]interface{})
+	output := dicomservices.CorsConfiguration{}
+
+	if v, ok := input["allowed_origins"]; ok {
+		output.Origins = utils.ExpandStringSlice(v.([]interface{}))
+	}
+
+	if v, ok := input["allowed_headers"]; ok {
+		output.Headers = utils.ExpandStringSlice(v.([]interface{}))
+	}
+
+	if v, ok := input["allowed_methods"]; ok {
+		output.Methods = utils.ExpandStringSlice(v.([]interface{}))
+	}
+
+	if v, ok := input["max_age_in_seconds"]; ok {
+		output.MaxAge = pointer.To(int64(v.(int)))
+	}
+
+	if v, ok := input["allow_credentials"]; ok {
+		output.AllowCredentials = pointer.To(v.(bool))
+	}
+
+	return &output
+}
+
+func flattenDicomServiceCorsConfiguration(input *dicomservices.CorsConfiguration) []interface{} {
+	outputList := make([]interface{}, 0)
+	if input == nil {
+		return outputList
+	}
+
+	output := make(map[string]interface{})
+	if input.AllowCredentials != nil {
+		output["allow_credentials"] = pointer.From(input.AllowCredentials)
+	}
```

Comment:
```suggestion
	output["allow_credentials"] = pointer.From(input.AllowCredentials)
```

Reply:
Updated the `flattenDicomServiceCorsConfiguration` function, which is used by the resource & data source